lib/python/rose/apps/ana_builtin/grepper.py

```diff
@@ -41,20 +45,34 @@ def run_command_and_check(self):
         self.command = self.command.format(*self.files)
         returncode, stdout, stderr = self.run_command(self.command)
         if returncode == 0:
-            self.parent.reporter(stdout, prefix="[INFO] ")
+            self.reporter(stdout, prefix="[INFO] ")
             self.passed = True
         else:
-            self.parent.reporter(stderr, prefix="[FAIL] ")
+            self.reporter(stderr, prefix="[FAIL] ")
+
+    def check_for_skip(self):
+        # If the users config options specified that the task should be
+        # ignored if all of its files were missing, set skipped attribute here
+        if self.skip_if_missing and len(self.files) > 0:
```

Comment:
I usually go with the call to `len` for readability and because of the effective in-built type checking that enforces... if `self.files` were ever accidentally _not_ a list/iterable then my version will raise a  `TypeError: object of type X has no len()`, whereas your version will continue (with the status of the check varying depending on what type `self.files` happens to be)...

Although I concede that in this instance we are in direct control of the contents of `self.files` so perhaps the above is a moot point